lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -604,6 +613,10 @@ export class NativeInteractionClient extends BaseInteractionClient {
             idTokenClaims
         );
 
+        if (accountInfo.nativeAccountId !== response.account.id) {
+            accountInfo.nativeAccountId = response.account.id;
```

Comment:
How do we get into this state and why is this not an error?

Reply:
The accountInfo is what we get from the baseAccount passed by the handleNativeResponse() method. The baseAccount comes from the buildAccountToCache method in the common ResponseHandler but that method overrides the embedded app's native account id that was passed in with the cached account (in this case it is the parent's cache). [This line specifically](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/e7e24e1b5d712c48d6ddc1374eacd7e7599bc18b/lib/msal-common/src/response/ResponseHandler.ts#L690C1-L691C1). Do we want to change that behaviour instead?